src/rgw/driver/rados/rgw_data_sync.cc

```diff
@@ -4521,12 +4521,6 @@ class RGWBucketSyncSingleEntryCR : public RGWCoroutine {
             goto done;
           }
 
-          // make sure versioned object only lands on versioned bucket and non-versioned object only lands on non-versioned bucket
-          if (key.instance.empty() == sync_pipe.dest_bucket_info.versioned()) {
```

Comment:
If I'm not mistaken, then bilogs created when versioning is suspended will have an empty instance but still have `RGW_BILOG_FLAG_VERSIONED_OP` set in bilogs, resulting in `entry->is_versioned()` being true when passing to `RGWBucketSyncSingleEntryCR()` eventually.

What about we change this to:
```cpp
if (versioned == sync_pipe.dest_bucket_info.versioned()) {
```

then we don't rely on instance.